keps/2941-DRA/README.md

```diff
@@ -0,0 +1,754 @@
+2. For DRAAdminAccess: This feature can only be enabled in admin namespace, therefore it should be skipped for being
+   counted against ClusterQuota. This is a different stance than Kubernetes ResourceQuota because kubernetes
+   ResourceQuota is namespace scoped. As a result, admin users can account for quota independent of user workloads.
+   On the contrary, with Kueue, since quota is part of ClusterQuota a cluster scoped object, admin workloads using
+   devices in admin namespace if counted against quota, will eat up quota meant for user workloads.
+3. For ResourceClaims with allocation mode `All`: worst-case scenario of the max number of devices that could be
+   allocated to a single claim will be used against quota.
+
+
+## Design Details
+
+A new feature gate DynamicResourceAllocation will be introduced, allowing users to test it in dev environments while
+making the changes dormant for production users. The following sections will explain the design in detail.
+
+### DynamicResourceAllocationConfig API
+
+```golang
+// DynamicResourceAllocationConfig is a singleton CRD that maps resource names todevice classes
+// used in ClusterQueue resource quotas. It is singleton as "default" is the only allowed named for the CRD instance in
+// Kueue namespace.
+type DynamicResourceAllocationConfig struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of DynamicResourceAllocationConfig
+    Spec DynamicResourceAllocationConfigSpec `json:"spec"`
+}
+
+// DynamicResourceAllocationConfigSpec defines the mappings between resource names and device classes
+type DynamicResourceAllocationConfigSpec struct {
+    // Resources is a list of mappings from resource name to device classes
+    // +listType=atomic
+    // +kubebuilder:validation:MaxItems=16
+    Resources []Resources `json:"resources"`
+}
+
+// Resources defines a mapping from a resource name to a list of device classes
+type Resources struct {
+    // Name is the resource name that will be referred to in ClusterQueue and Workload status
+    Name corev1.ResourceName `json:"name"`
+
+    // DeviceClassNames lists the names of all the device classes that will count against
+    // the quota defined for this resource name
+    // +listType=atomic
+    // +kubebuilder:validation:MaxItems=8
+    DeviceClassNames []corev1.ResourceName `json:"deviceClassNames"`
+}
+```
+
+The ResourceFlavor spec will have a field called dynamicResource. The cluster admin has to define the list of
+deviceClasses that will be represented by a resource name and can be used to define quotas in cluster queue. With this
+a ClusterQueue with nominalQuota can be defined as follows:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: DynamicResourceAllocationConfig
+metadata:
+  name: "default"  # Fixed name - singleton CR, only one instance allowed
+  namespace: "kube-system"
+spec:
+  dynamicResources:
```

Comment:
```suggestion
  resources:
```
to align with the API

Reply:
Good catch, updated all such instances, found 4 more further down the doc that needed to be fixed.

Reply:
thx, feel free to resolve